### ISSUE TITLE
no-redeclare false positives in class methods

### ISSUE THREAD

**Issue by lucacasonato** [denoland/deno#12032]
### Lint Name

`no-redeclare`

### Code Snippet

```ts
export class X {
  y() {
    const _res = "foo";
  }
  z() {
    const _res = "foo";
  }
}
```

### Expected Result

No `no-redeclare` hit.

### Actual Result

`no-redeclare` triggers on `const _res` in the `z()` method.

### Version

deno 1.13.2+b31dad8

**Comment by lucacasonato** [comment]
cc @dsherret

**Comment by bartlomieju** [comment]
The problem is described in https://github.com/denoland/deno_ast/issues/8

**Comment by lucacasonato** [comment]
Fixed in #11988

**Comment by lucacasonato** [comment]
Maybe adding the case in the OP as an integration test would be good tho.
